### PR TITLE
[SPARK-23355][SQL][DOC][FOLLOWUP] Add migration doc for TBLPROPERTIES

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
## What changes were proposed in this pull request?

In Apache Spark 2.4, [SPARK-23355](https://issues.apache.org/jira/browse/SPARK-23355) fixes a bug which ignores table properties during convertMetastore for tables created by STORED AS ORC/PARQUET.

For some Parquet tables having table properties like TBLPROPERTIES (parquet.compression 'NONE'), it was ignored by default before Apache Spark 2.4. After upgrading cluster, Spark will write uncompressed file which is different from Apache Spark 2.3 and old.

This PR adds a migration note for that.

## How was this patch tested?

N/A

### CODE REVIEW

File: docs/sql-programming-guide.md [hunk 1]
```diff
@@ -1812,6 +1812,8 @@ working with timestamps in `pandas_udf`s to get the best performance, see
...
+  - In version 2.3 and earlier, Spark converts Parquet Hive tables by default but ignores table properties like `TBLPROPERTIES (parquet.compression 'NONE')`. This happens for ORC Hive table properties like `TBLPROPERTIES (orc.compress 'NONE')` in case of `spark.sql.hive.convertMetastoreOrc=true`, too. Since Spark 2.4, Spark supports Parquet/ORC specific table properties while converting Parquet/ORC Hive tables. As an example, `CREATE TABLE t(id int) STORED AS PARQUET TBLPROPERTIES (parquet.compression 'NONE')` would generate Snappy parquet files during insertion in Spark 2.3, and in Spark 2.4, the result would be uncompressed parquet files.
```
Comment (cloud-fan) [inline docs/sql-programming-guide.md]: `Spark supports Parquet/ORC specific table properties while converting ..`

supports -> respects

**Comment by cloud-fan** [comment]
LGTM

File: docs/sql-programming-guide.md (hunk 1 above)
Comment (dongjoon-hyun) [reply docs/sql-programming-guide.md]: Thanks. It's updated.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon and @cloud-fan .

**Pull request closed without merging**